lib/cisco_node_utils/interface.rb

```diff
@@ -499,13 +475,11 @@ def default_mtu
     end
 
     def speed
-      config_get('interface', 'speed', name: @name)
+      val = config_get('interface', 'speed', name: @name)
+      val == 'auto' ? val : val.to_i
```

Comment:
1. Do you really need the `to_i`? Can we just handle it as a string? Or is this fixing a particular problem?
2. If val is nil then val.to_i == 0